proxy/src/main/java/com/wavefront/agent/listeners/tracing/CustomTracingPortUnificationHandler.java

```diff
@@ -0,0 +1,239 @@
+        null : preprocessorSupplier.get();
+    String[] messageHolder = new String[1];
+
+    // transform the line if needed
+    if (preprocessor != null) {
+      message = preprocessor.forPointLine().transform(message);
+
+      if (!preprocessor.forPointLine().filter(message, messageHolder)) {
+        if (messageHolder[0] != null) {
+          handler.reject((Span) null, messageHolder[0]);
+        } else {
+          handler.block(null, message);
+        }
+        return;
+      }
+    }
+
+    List<Span> output = Lists.newArrayListWithCapacity(1);
+    try {
+      decoder.decode(message, output, "dummy");
+    } catch (Exception e) {
+      handler.reject(message, formatErrorMessage(message, e, ctx));
+      return;
+    }
+
+    for (Span object : output) {
+      if (preprocessor != null) {
+        preprocessor.forSpan().transform(object);
+        if (!preprocessor.forSpan().filter(object, messageHolder)) {
+          if (messageHolder[0] != null) {
+            handler.reject(object, messageHolder[0]);
+          } else {
+            handler.block(object);
+          }
+          return;
+        }
+      }
+
+      // check whether error span tag exists.
+      boolean sampleError = alwaysSampleErrors && object.getAnnotations().stream().anyMatch(
+          t -> t.getKey().equals(ERROR_SPAN_TAG_KEY) && t.getValue().equals(ERROR_SPAN_TAG_VAL));
+      if (sampleError || sample(object)) {
+        handler.report(object);
+
+        // report converted metrics/histograms from the span
+        String applicationName = NULL_TAG_VAL;
+        String serviceName = NULL_TAG_VAL;
+        String cluster = NULL_TAG_VAL;
+        String shard = NULL_TAG_VAL;
+        String componentTagValue = NULL_TAG_VAL;
+        String isError = "false";
+        if (wfInternalReporter != null) {
+          List<Annotation> annotations = object.getAnnotations();
+          for (Annotation annotation : annotations) {
+            switch (annotation.getKey()) {
+              case APPLICATION_TAG_KEY:
+                applicationName = annotation.getValue();
+                continue;
+              case SERVICE_TAG_KEY:
+                serviceName = annotation.getValue();
+              case CLUSTER_TAG_KEY:
+                cluster = annotation.getValue();
+                continue;
+              case SHARD_TAG_KEY:
+                shard = annotation.getValue();
+                continue;
+              case COMPONENT_TAG_KEY:
+                componentTagValue = annotation.getValue();
+                break;
+              case ERROR_SPAN_TAG_KEY:
+                isError = annotation.getValue();
+                break;
+            }
+          }
+
+          if (applicationName.equals(NULL_TAG_VAL) || serviceName.equals(NULL_TAG_VAL)) {
+            logger.warning("Ingested spans discarded because span application/service name is " +
+                "missing.");
+            discardedSpans.inc();
```

Comment:
seems like the span is already reported above, so this code is not doing what you intended.